karaf/init/src/main/java/org/apache/brooklyn/launcher/osgi/OsgiLauncher.java

```diff
@@ -23,36 +26,78 @@
 import org.apache.brooklyn.core.internal.BrooklynProperties;
 import org.apache.brooklyn.core.mgmt.persist.PersistMode;
 import org.apache.brooklyn.launcher.common.BasicLauncher;
+import org.apache.brooklyn.launcher.common.BrooklynPropertiesFactoryHelper;
 import org.apache.brooklyn.rest.BrooklynWebConfig;
 import org.apache.brooklyn.rest.security.provider.BrooklynUserWithRandomPasswordSecurityProvider;
+import org.apache.brooklyn.util.exceptions.Exceptions;
 import org.apache.brooklyn.util.javalang.Threads;
+import org.apache.brooklyn.util.text.Strings;
 import org.apache.brooklyn.util.time.Duration;
+import org.osgi.framework.Constants;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Initializer for brooklyn-core when running in an OSGi environment.
- *
- * Temporarily here; should be totally contained in blueprint beans' init-methods.
  */
 public class OsgiLauncher extends BasicLauncher<OsgiLauncher> {
 
     private static final Logger LOG = LoggerFactory.getLogger(OsgiLauncher.class);
+    public static final String BROOKLYN_CONFIG_PID = "brooklyn";
```

Comment:
I don't think we need to feel strictly constrained to using package style naming, 'org.apache.brooklyn'.  It's certainly appropriate if the configuration is for a module, where the package style is needed to distinguish itself from others that might have the same basename, e.g. x.y.z.core.security.cfg versus x.y.z.software.base.security.cfg. In this case however we're configuring the whole product, rather than a module. I think we can be a bit flexible here for the convenience of the simpler file name.